src/Sarif.Viewer.VisualStudio/ErrorList/SarifAbsentResultsTableDataSource.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.TableManager;
+
+namespace Microsoft.Sarif.Viewer.ErrorList
+{
+    // A "stub" data source whose sole purpose is to cause the Category column
+    // to be displayed. It never actually adds entries to the Error List.
+    //
+    // For an explanation of why these this source is necessary, see the comment
+    // near the top of SarifResultTableEntry.cs
+    internal class SarifAbsentResultsTableDataSource : SarifTableDataSourceBase
+    {
+        private static SarifAbsentResultsTableDataSource _instance;
+
+        private SarifAbsentResultsTableDataSource()
+        {
+            if (!SarifViewerPackage.IsUnitTesting)
+            {
+#pragma warning disable VSTHRD108 // Assert thread affinity unconditionally
+                ThreadHelper.ThrowIfNotOnUIThread();
+#pragma warning restore VSTHRD108 // Assert thread affinity unconditionally
+
+                this.Initialize(SarifResultTableEntry.AbsentResultColumns);
+            }
+        }
+
+        public static SarifAbsentResultsTableDataSource Instance
+        {
+            get
+            {
+                if (_instance == null)
+                {
+                    _instance = new SarifAbsentResultsTableDataSource();
```

Comment:
>_instance [](start = 20, length = 9)

You aren't using the assignment coalescing operator I assume because we haven't upgraded to 8.0 yet? If so: [https://github.com/microsoft/sarif-visualstudio-extension/issues/314](https://github.com/microsoft/sarif-visualstudio-extension/issues/314). Low priority update.

Reply:
Yes, that's the reason.

---
In reply to: [538753105](https://github.com/microsoft/sarif-visualstudio-extension/pull/313#discussion_r538753105) [](ancestors = 538753105)